Bentheburrito/sacastats
# Outfit Polls Groundwork

This is the groundwork/basic functionality for outfit polls. It allows logged-in users to create a poll with text-based or multiple choice poll items. Then, users (logged in or not) can view the poll and vote for particular items.

This is still pretty far from being done. The poll-management page for the poll owner still needs to be added. There are also probably some bugs/validation issues to be fixed, as I learned how to use nested schemas and how to use them with Ecto/forms/LiveView all at once.

## To test
1. Log in with Discord, then go to `/outfit/poll/create`
2. Add some poll items, fill in the form contents, click "Create Poll"
3. The poll should be created and you'll be redirected to `/outfit/poll/POLL_ID`
4. Since the management page doesn't exist yet, the owner can just vote on the poll for now.
5. You can open private browser window(s) to cast votes as anonymous users, or as logged-in users if you use other Discord accounts.
6. You can see the data stored in the DB with `SacaStats.Repo.get(SacaStats.Poll, POLL_ID_HERE) |> SacaStats.Repo.preload([:text_items, :multi_choice_items])`
